adaptors_task/student/include/impl.h

```diff
@@ -11,4 +14,42 @@
  *
  * @return true if expression is valid
  */
-bool isValid(const std::string& source);
+bool isValid(const std::string& source){
+    stack <string> openingbrackets;
+    for (const char ch : source){
+        switch (ch)
+        {
+        case '(':
+        case '{':
+        case '[':
+            openingbrackets.push(&ch);
+            break;
+
+        case ')':
+            if (openingbrackets.empty() || openingbrackets.top() != "("){
+                return false;
+            }
+            openingbrackets.pop();
+            break;
+        
+        case '}':
+            if (openingbrackets.empty() || openingbrackets.top() != "{"){
+                return false;
+            }
+            openingbrackets.pop();
+            break;
+        
+        case ']':
+            if (openingbrackets.empty() || openingbrackets.top() != "["){
```

Comment:
we'll need to repeat this each time we introduce/change a bracket type, so please let's store this info in container from impl.cpp